### PR TITLE
fix: clear substantive new-code Sonar findings

### PR DESCRIPTION
**Pull request by Robbie-Palmer**
## Summary

- refactor the three new-code cognitive-complexity hotspots into focused helpers
- replace nested ternaries and strengthen Bash condition/parameter handling
- use native JSX semantics for goal progress, checkbox copy, and recipe steps
- add regression coverage for timer actions and native goal progress

## Verification

- `MISE_EXPERIMENTAL=1 mise //ui:check` (566 tests)
- Bash syntax checks and ShellCheck for all three changed scripts

Refs #732

### CODE REVIEW

File: ui/components/recipes/recipe-content.tsx [hunk 1]
```diff
@@ -615,13 +615,9 @@ export function RecipeContent({
...
-        {/* Real ordered list whose visible numeral is decorative generated
-            content via a CSS counter (aria-hidden), so screen readers rely on
-            list position. role="list" is redundant per spec but kept on purpose:
-            Safari/VoiceOver drop list semantics when the marker is removed with
-            list-style:none, and reasserting the role restores them. */}
-        {/* biome-ignore lint/a11y/noRedundantRoles: intentional Safari/VoiceOver list-semantics workaround (see above) */}
-        <ol role="list" className="rt-method-steps list-none p-0 m-0">
+        {/* The visible numeral is decorative generated content via a CSS
+            counter, while the ordered list retains the document semantics. */}
+        <ol className="rt-method-steps list-none p-0 m-0">
```
[the remark this answers is not in the document]
Comment (Robbie-Palmer) [reply ui/components/recipes/recipe-content.tsx]: Addressed in 7b9a75c without reintroducing redundant ARIA. The list now retains a native decimal marker and only makes the marker text transparent, so Safari/VoiceOver keeps the ordered-list semantics while the existing CSS counter remains the visible numeral. I also added a regression test preventing `list-none` from returning.

File: ui/components/assettracker/net-worth-chart.tsx [hunk 2]
```diff
@@ -120,12 +120,9 @@ export function NetWorthChart({ data }: NetWorthChartProps) {
...
       : null;
-  const rangeLabel =
-    rangeYears == null
-      ? "all time"
-      : rangeYears === 1
-        ? "past year"
-        : `past ${rangeYears} years`;
+  let rangeLabel = `past ${rangeYears} years`;
+  if (rangeYears == null) rangeLabel = "all time";
+  if (rangeYears === 1) rangeLabel = "past year";
```
[the remark this answers is not in the document]
Comment (Robbie-Palmer) [reply ui/components/assettracker/net-worth-chart.tsx]: Applied in 7b9a75c. The label now uses an explicit if/else-if/else assignment, avoiding both the nested ternary and the discarded interpolated strings.

File: ui/components/recipes/recipe-content.tsx [hunk 3]
```diff
@@ -615,13 +615,10 @@ export function RecipeContent({
...
-        {/* Real ordered list whose visible numeral is decorative generated
-            content via a CSS counter (aria-hidden), so screen readers rely on
-            list position. role="list" is redundant per spec but kept on purpose:
-            Safari/VoiceOver drop list semantics when the marker is removed with
-            list-style:none, and reasserting the role restores them. */}
-        {/* biome-ignore lint/a11y/noRedundantRoles: intentional Safari/VoiceOver list-semantics workaround (see above) */}
-        <ol role="list" className="rt-method-steps list-none p-0 m-0">
+        {/* Keep a native decimal marker so Safari/VoiceOver retain the list
+            semantics. Its text is transparent because the visible numeral is
+            decorative generated content from the CSS counter below. */}
```
[the remark this answers is not in the document]
Comment (Robbie-Palmer) [reply ui/components/recipes/recipe-content.tsx:620]: This is already addressed in the current 7b9a75c head: the `<ol>` no longer uses `list-none`; it uses `list-decimal marker:text-transparent`. Safari therefore retains a native decimal list marker and its accessibility-tree semantics, while only the marker text is visually transparent. The adjacent comment explains this, and the new regression test prevents `list-none` from returning.

File: ui/components/assettracker/net-worth-chart.tsx [hunk 4]
```diff
@@ -120,12 +120,14 @@ export function NetWorthChart({ data }: NetWorthChartProps) {
...
       : null;
-  const rangeLabel =
-    rangeYears == null
-      ? "all time"
-      : rangeYears === 1
-        ? "past year"
-        : `past ${rangeYears} years`;
+  let rangeLabel: string;
+  if (rangeYears == null) {
+    rangeLabel = "all time";
```
[the remark this answers is not in the document]
Comment (Robbie-Palmer) [reply ui/components/assettracker/net-worth-chart.tsx:125]: This is already addressed in the current 7b9a75c head. `rangeLabel` is declared without an initializer and assigned through an explicit if/else-if/else block, so neither `past null years` nor `past 1 years` is allocated.
